source/common/res/features/budget-to-goal/main.js

```diff
@@ -0,0 +1,163 @@
+(function poll() {
+  if (typeof ynabToolKit !== 'undefined' && ynabToolKit.pageReady === true) {
+    ynabToolKit.budgetBalanceToGoal = (function () {
+      return {
+        budgetView: ynab.YNABSharedLib.getBudgetViewModel_AllBudgetMonthsViewModel()._result,
+
+        invoke() {
+          let categories = ynabToolKit.budgetBalanceToGoal.getCategories();
+          let categoryName = ynabToolKit.budgetBalanceToGoal.getInspectorName();
+          let masterCategoryViewId = $('ul.is-checked').prevAll('ul.is-master-category').attr('id');
+          let masterCategory = ynabToolKit.shared.getEmberView(masterCategoryViewId).get('data');
+          let masterCategoryId = masterCategory.get('categoryId');
+
+          categories.forEach(function (f) {
+            if (f.name === categoryName && f.masterCategoryId === masterCategoryId) {
+              ynabToolKit.budgetBalanceToGoal.updateInspectorButton(f);
+
+              return false;
+            }
+          });
+        },
+
+        observe(changedNodes) {
+          console.log(changedNodes);
+          if (changedNodes.has('budget-inspector-goals')) {
+            // In theory, if there's a 'budget-inspector-goals' classed node, then the user is or has changed the goal.
+            // However, there seems to be some disconnect between this event and when YNAB updates the values.
+            setTimeout(function () {
+              ynabToolKit.budgetBalanceToGoal.invoke();
+            }, 60000);
+          } else if (changedNodes.has('budget-inspector')) {
+            // User has changed selected budget item or changed budgeted amount.
+            ynabToolKit.budgetBalanceToGoal.invoke();
+          }
+        },
+
+        addBudgetVersionIdObserver() {
+          let applicationController = ynabToolKit.shared.containerLookup('controller:application');
+          applicationController.addObserver('budgetVersionId', function () {
+            Ember.run.scheduleOnce('afterRender', this, resetBudgetView);
+          });
+
+          function resetBudgetView() {
+            ynabToolKit.budgetBalanceToGoal.budgetView = null;
+          }
+        },
+
+        getCategories() {
+          // After using Budget Quick Switch, budgetView needs to be reset to the new budget. The try catch construct is necessary
+          // because this function can be called several times during the budget switch process.
+          if (ynabToolKit.budgetBalanceToGoal.budgetView === null) {
+            try {
+              ynabToolKit.budgetBalanceToGoal.budgetView = ynab.YNABSharedLib.getBudgetViewModel_AllBudgetMonthsViewModel()._result;
```

Comment:
We're trying to get away from using `._result` because although we've not seen it go wrong yet, it's depending on an implementation detail of a promise library that we don't control. We're trying to find a good option around it, stay up to date or chip into the conversation here: https://github.com/toolkit-for-ynab/toolkit-for-ynab/pull/727